src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -502,6 +502,142 @@ def query_for_quota_alert(self):
             for dataset in self.query()
         ]
 
+    def common_load_dataset_checks(self, ds):
+        self.common_encryption_checks(ds)
+        if ds.key_loaded:
+            raise CallError(f'{id} key is already loaded')
+
+    def common_encryption_checks(self, ds):
+        if not ds.encrypted:
+            raise CallError(f'{id} is not encrypted')
+
+    @accepts(
+        Str('id'),
+        Dict(
+            'load_key_options',
+            Bool('mount', default=False),
+            Bool('recursive', default=False),
+            Any('key', default=None, null=True),
+            Str('key_location', default=None, null=True),
+        ),
+    )
+    def load_key(self, id, options):
+        mount_ds = options.pop('mount')
+        recursive = options.pop('recursive')
+        try:
+            with libzfs.ZFS() as zfs:
+                ds = zfs.get_dataset(id)
+                self.common_load_dataset_checks(ds)
+                ds.load_key(**options)
+        except libzfs.ZFSException as e:
+            self.logger.error(f'Failed to load key for {id}', exc_info=True)
+            raise CallError(f'Failed to load key for {id}: {e}')
+        else:
+            if mount_ds:
+                self.mount(id, {'recursive': recursive})
+
+    @accepts(Str('name'), List('params', default=[]))
+    @job(transient=True)
+    def bulk_process(self, job, name, params):
+        f = getattr(self, name, None)
+        if not f:
+            raise CallError(f'{name} method not found in zfs.dataset')
+
+        statuses = []
+        for i in params:
+            result = error = None
+            try:
+                result = f(*i)
+            except Exception as e:
+                error = str(e)
+            finally:
+                statuses.append({'result': result, 'error': error})
+
+        return statuses
+
+    @accepts(
+        Str('id'),
+        Dict(
+            'check_key',
+            Any('key', default=None, null=True),
+            Str('key_location', default=None, null=True),
+        )
+    )
+    def check_key(self, id, options):
+        """
+        Returns `true` if the `key` is valid, `false` otherwise.
+        """
+        try:
+            with libzfs.ZFS() as zfs:
+                ds = zfs.get_dataset(id)
+                self.common_encryption_checks(ds)
+                return ds.check_key(**options)
+        except libzfs.ZFSException as e:
+            self.logger.error(f'Failed to check key for {id}', exc_info=True)
+            raise CallError(f'Failed to check key for {id}: {e}')
+
+    @accepts(
+        Str('id'),
+        Dict(
+            'unload_key_options',
+            Bool('recursive', default=False),
+            Bool('force_umount', default=False),
+            Bool('umount', default=False),
```

Comment:
Nope, it is not possible. However i have tried to be consistent here with how ZFS operates exposing option to `umount` it but not doing it implicitly so that the developer is aware this needs to be done. In case this is not done, an exception will be raised